Example/Tests/ViewModels/ConnectionsViewModelSpec.swift

```diff
@@ -0,0 +1,206 @@
+//  WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+//  General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+//  For the additional permissions granted for Salt Edge Authenticator
+//  under Section 7 of the GNU General Public License see THIRD_PARTY_NOTICES.md
+//
+
+import Quick
+import Nimble
+import SEAuthenticator
+@testable import SEAuthenticatorCore
+
+private final class MockConnectionsDelegate: ConnectionsEventsDelegate {
+    var showEditConnectionAlertCall: Bool = false
+    var showSupportCall: Bool = false
+    var deleteConnectionCall: Bool = false
+    var reconnectCall: Bool = false
+    var consentsPressedCall: Bool = false
+    var updateViewsCall: Bool = false
+    var addPressedCall: Bool = false
+    var presentErrorCall: Bool = false
+    var showNoInternetConnectionAlertCall: Bool = false
+    var showDeleteConfirmationAlertCall: Bool = false
+    
+    func showEditConnectionAlert(placeholder: String, completion: @escaping (String) -> ()) {
+        showEditConnectionAlertCall = true
+    }
+    
+    func showSupport(email: String) {
+        showSupportCall = true
+    }
+    
+    func deleteConnection(completion: @escaping () -> ()) {
+        deleteConnectionCall = true
+    }
+    
+    func reconnect(by id: String) {
+        reconnectCall = true
+    }
+    
+    func consentsPressed(connectionId: String, consents: [SEConsentData]) {
+        consentsPressedCall = true
+    }
+    
+    func updateViews() {
+        updateViewsCall = true
+    }
+    
+    func addPressed() {
+        addPressedCall = true
+    }
+    
+    func presentError(_ error: String) {
+        presentErrorCall = true
+    }
+    
+    func showNoInternetConnectionAlert(completion: @escaping () -> Void) {
+        showNoInternetConnectionAlertCall = true
+    }
+    
+    func showDeleteConfirmationAlert(completion: @escaping () -> Void) {
+        showDeleteConfirmationAlertCall = true
+    }
+}
+
+final class ConnectionsViewModelSpec: BaseSpec {
+    private var viewModel: ConnectionsViewModel!
+    private var networkSpy: MockConnectable!
+    
+    override func spec() {
+        beforeEach {
+            self.networkSpy = MockConnectable()
+            self.viewModel = ConnectionsViewModel(reachabilityManager: self.networkSpy)
+            ConnectionRepository.deleteAllConnections()
+        }
+        afterEach {
+            self.viewModel = nil
+            self.networkSpy = nil
+        }
+                
+        describe("count") {
+            it("should return count of connections") {
+                let connection = Connection()
+                connection.status = "active"
+
+                ConnectionRepository.save(connection)
+            
+                expect(self.viewModel.count).to(equal(1))
```

Comment:
rm `self.`

Reply:
Reference to property 'viewModel' in closure requires explicit use of 'self' to make capture semantics explicit

Reply:
this is not a closure